doc/rados/troubleshooting/troubleshooting-osd.rst

```diff
@@ -150,88 +162,134 @@ If you start your cluster and an OSD won't start, check the following:
+  `OS recommendations`_ and the release notes for each Ceph version
   to ensure you have addressed any issues related to your kernel.
 
-- **Segment Fault:** If there is a segment fault, turn your logging up
-  (if it is not already), and try again. If it segment faults again,
-  contact the ceph-devel email list and provide your Ceph configuration
-  file, your monitor output and the contents of your log file(s).
-
+- **Segment Fault:** If there is a segment fault, increase log levels
+  and start the problematic daemon(s) again. If segment faults recur,
+  search the Ceph bug tracker `https://tracker.ceph/com/projects/ceph <https://tracker.ceph.com/projects/ceph/>`_
+  and the ``dev`` and ``ceph-users`` mailing list archives `https://ceph.io/resources <https://ceph.io/resources>`_.
+  If this is truly a new and unique
+  failure, post to the ``dev`` email list and provide the specific Ceph
+  release being run, ``ceph.conf`` (with secrets XXX'd out),
+  your monitor status output and excerpts from your log file(s).
 
 
 An OSD Failed
 -------------
 
-When a ``ceph-osd`` process dies, the monitor will learn about the failure
-from surviving ``ceph-osd`` daemons and report it via the ``ceph health``
-command::
+When a ``ceph-osd`` process dies, surviving ``ceph-osd`` daemons will report
+to the mons that it appears down, which will in tur surface the new status
+via the ``ceph health`` command::
 
 	ceph health
 	HEALTH_WARN 1/3 in osds are down
 
-Specifically, you will get a warning whenever there are ``ceph-osd``
-processes that are marked ``in`` and ``down``.  You can identify which
-``ceph-osds`` are ``down`` with::
+Specifically, you will get a warning whenever there are OSDs marked ``in``
+and ``down``.  You can identify which  are ``down`` with::
 
 	ceph health detail
 	HEALTH_WARN 1/3 in osds are down
 	osd.0 is down since epoch 23, last address 192.168.106.220:6800/11080
 
-If there is a disk
+or ::
+
+	ceph osd tree down
+
+
+If there is a drive
 failure or other fault preventing ``ceph-osd`` from functioning or
-restarting, an error message should be present in its log file in
+restarting, an error message should be present in its log file under
 ``/var/log/ceph``.
 
-If the daemon stopped because of a heartbeat failure, the underlying
-kernel file system may be unresponsive. Check ``dmesg`` output for disk
-or other kernel errors.
+If the daemon stopped because of a heartbeat failure or ``suicide timeout``,
+the underlying drive or filesystem may be unresponsive. Check ``dmesg``
+and `syslog`  output for drive or other kernel errors.  You may need to
+specify something like ``dmesg -T`` to get timestamps, otherwise it's
+easy to mistake old errors for new.
 
 If the problem is a software error (failed assertion or other
-unexpected error), it should be reported to the `ceph-devel`_ email list.
+unexpected error), search the archives and tracker as above, and
+reported to the `ceph-devel`_ email list if there's no clear fix or
```

Comment:
s/reported/report it/

Reply:
Ack.  As I reworked this bit and others like it I wanted to encourage the reader to look for existing answers before going to the list or tracker.   Traffic on the list is already high; we don't need to make it worse.